### PR TITLE
chore(package.xml): bump version to 0.39.0

### PR DESCRIPTION
**Pull request by mitsudome-r**
## Description

This PR updates package.xml and CHANGELOG.rst in the `main` branch according to the release done in the `humble` branch.

We will be taking different release process from 0.40.0, but we need this update to be done for 0.39.0.

## Related links

*  https://github.com/autowarefoundation/autoware.universe/pull/9435
* https://github.com/orgs/autowarefoundation/discussions/5518

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Comment by mitsudome-r** [comment]
I have noticed the ticket links were referring to @youtalk's fork so I have fixed in this commit: 2b0076e0ee5684291cd38c0517d3567ed04f1e1c. 

I have also created the fix PR to humble branch as well. https://github.com/autowarefoundation/autoware.universe/pull/9588

File: common/autoware_time_utils/CHANGELOG.rst [hunk 1]
```diff
@@ -1,6 +1,24 @@
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (youtalk) [inline common/autoware_time_utils/CHANGELOG.rst:1]: This diff is something weird.

**Review by youtalk — APPROVED** [review]
I found 1 suggestion but it's not critical.

**Review by sasakisasaki — APPROVED** [review]
Looks good!

Reasons:
- This is the PR for version update from `0.38.0` to `0.39.0`
- I double checked if there is no any missing update by following command:
```
grep -R '<version>0.38.0</version>' ./
```
- It seems in the `scenario_simulator_v2_adaptor`, the `CHANGELOG.rst` is missing ([related link](https://github.com/autowarefoundation/autoware.universe/pull/9501/files)). But this missing file would be created in the next release timing. Thus I guess this is not an issue in this PR.

File: common/autoware_time_utils/CHANGELOG.rst (hunk 1 above)
Comment (mitsudome-r) [reply common/autoware_time_utils/CHANGELOG.rst]: Thanks for pointing out.
Fixed in 873a6bb497528e4ccf210182b1d6bc2cb1ba0eb1

**Pull request merged**